src/main/java/SubwayNetwork.java

```diff
@@ -0,0 +1,76 @@
+import com.sun.org.apache.bcel.internal.generic.FSUB;
+import jdk.internal.jline.internal.TestAccessible;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class SubwayNetwork {
+
+    private Map<Integer, SubwayStation.Station> stations;
+
+    public List<SubwayStation.Station> findShortestPath(SubwayStation.Station origin, SubwayStation.Station destination) throws IOException {
+        List<SubwayStation.Station> visitedStations = dijkstraAlgorithm(origin, destination);
+        return tracePath(visitedStations, origin, destination);
+    }
+
+    private List<SubwayStation.Station> dijkstraAlgorithm (SubwayStation.Station origin, SubwayStation.Station destination) throws IOException {
+        Converter converter = new Converter();
+        List<SubwayStation.Station> unvisitedStations = converter.getStationsList();
+        List<SubwayStation.Station> visitedStations = new ArrayList<>();
+
+        for (SubwayStation.Station station : unvisitedStations) {
+            station.getProperties().setVisited(false);
+            station.getProperties().setCurrent(false);
+            station.getProperties().setDistance(Integer.MAX_VALUE);
+        }
+
+        SubwayStation.Station currentStation = unvisitedStations.get(unvisitedStations.indexOf(origin));
```

Comment:
You have `origin`
You use origin to get the index in `unvisitedStations`
You then use the index to get the Station in `unvisitedStations`... which is `origin`